lnrpc/signrpc/signer_server.go

```diff
@@ -547,3 +550,84 @@ func (s *Server) DeriveSharedKey(_ context.Context, in *SharedKeyRequest) (
 
 	return &SharedKeyResponse{SharedKey: sharedKeyHash[:]}, nil
 }
+
+// getSharedKeyHashFromPubKey uses the raw public key to do the ECDH. Because
+// the function ECDH assumes a valid key family, which we cannot tell from the
+// raw public key, we need to try all the key families to find the right one.
+//
+// NOTE: The wrost case will derive 1,000,000 keys to find out whether there is
+// a match against the given public key:
+//   derive times = len(VersionZeroKeyFamilies) x keychain.MaxKeyRangeScan
+func (s *Server) getSharedKeyHashFromPubKey(
```

Comment:
This is already done in https://github.com/lightningnetwork/lnd/blob/master/keychain/btcwallet.go#L250. We just need to expect the caller to always specify the family.

Reply:
removed.